users/faq-parts/general.rst

```diff
@@ -48,6 +48,37 @@ The following are *not* synchronized;
 -  Devices, FIFOs, and other specials (ignored)
 -  Sparse file sparseness (will become sparse, when supported by the OS & filesystem)
 
+What does syncing mean for links?
+---------------------------------
+
+As mentioned above, `symbolic links
+<https://en.wikipedia.org/wiki/Symbolic_link>`__ are synced but not followed
+(except on Windows, where they're not synced). To see what this means, consider
+the situation where you have a symbolic link at ``./Sync/symlink`` that points
+to ``../some-file.txt``.  If Syncthing syncs the ``./Sync`` directory, then it
+will sync ``./Sync/symlink`` *as a link* – that is, Syncthing will replicate the
+link on every synced device. But, because ``../some-file.txt`` isn't part of the
+synced directory, Syncthing won't sync ``some-file.txt`` (that's what "not
+followed" means).  Thus, the ``./Sync/symlink`` file will point to whatever
+(non-synced) file is at ``../some-file.txt`` on any particular device.  If no
+such file exists, ``./Sync/symlink`` will be a broken symlink.
+
+The situation for `hard links <https://en.wikipedia.org/wiki/Hard_link>`__ is
+nearly reversed: Syncthing does not sync hard links but *does* sync the
+underlying file (which is implied by the nature of hard links, which effectively
+are the file).  For example, if you have a hard link at ``./Sync/hardlink`` that
+points to a file at ``../some-file.txt``, then a sync of the ``./Sync``
+directory will replicate the contents of ``../some-file.txt`` to
+`./Sync/hardlink` on every device.  But it will *not* sync the fact that
+``./Sync/hardlink`` was a link on the original device – on other devices,
+``./Sync/hardlink`` will be completely independent from ``../some-file.txt`` (if
+that file exists).
+
+All of this results in a general rule of thumb: if Syncthing is syncing both a
+link and its target, it's better to use a symbolic link.  However, if the link
+points to a file that is *not* in a synced directory and you want to sync the
+contents of that file, you should use a hard link instead.
```

Comment:
I don't think it's appropriate to give advice on what type of links a user should create, from the perspective of the file synchronization software.  Symlinks and hardlinks are fundamentally different concepts despite the similar names.  Symlinks work on a path level, while hardlinks work on the inodes (basically).

We should just state that Syncthing does not respect the fact that two or more files have a hardlink relationship, it is not preserved when syncing to other devices.

The hard-link example here talks about the situation where two links (file names) point to the same file, one inside and one outside the synced folder.  However the explanation is just as relevant for the case when both are inside, since this obviously will double the space requirement on receiving devices.